File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/asset/dto/AssetDto.java
package com.side.tiggle.domain.asset.dto;

import com.side.tiggle.domain.asset.model.Asset;
import lombok.*;
import lombok.experimental.SuperBuilder;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@SuperBuilder
public class AssetDto {

    private Long id;
    private String name;
    private boolean defaults;

    public static AssetDto fromEntity(Asset asset) {
        return AssetDto.builder()
                .id(asset.getId())
                .name(asset.getName())
                .defaults(asset.isDefaults())
                .build();
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/asset/dto/AssetDto.kt
package com.side.tiggle.domain.asset.dto

import com.side.tiggle.domain.asset.model.Asset

data class AssetDto(
    val name: String,
    val defaults: Boolean
) {

    fun toEntity(): Asset {
        return Asset(
            name = this.name,
            defaults = this.defaults
        )
    }
    companion object {
        fun fromEntity(asset: Asset): AssetDto {
            return AssetDto(
                name = asset.name,
                defaults = asset.defaults
            )
        }
    }
}